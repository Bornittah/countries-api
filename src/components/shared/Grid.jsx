import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import { FaArrowAltCircleRight } from 'react-icons/fa';

const Grid = ({ title, number }) => (
  <Link to={`/continent/${title}`} className="link">
    <div className="grid">
      <FaArrowAltCircleRight />
      <div className="data-title">
        <h3>{title}</h3>
        <p>
          <span>{number}</span>
          <span>Countries</span>
        </p>
      </div>
    </div>
  </Link>
);

Grid.defaultProps = {
  title: null,
  number: null,
};

Grid.propTypes = {
  title: PropTypes.string,
  number: PropTypes.number,
};

export default Grid;
