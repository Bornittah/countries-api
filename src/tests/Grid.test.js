import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Grid from '../components/shared/Grid';

describe('Testing Grid component', () => {
  it('Testing with snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <Grid />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
