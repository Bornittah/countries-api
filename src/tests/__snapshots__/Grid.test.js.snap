// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Grid component Testing with snapshot 1`] = `
<div
  className="grid"
>
  <a
    className="link"
    href="/continent/null"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zM140 300h116v70.9c0 10.7 13 16.1 20.5 8.5l114.3-114.9c4.7-4.7 4.7-12.2 0-16.9l-114.3-115c-7.6-7.6-20.5-2.2-20.5 8.5V212H140c-6.6 0-12 5.4-12 12v64c0 6.6 5.4 12 12 12z"
      />
    </svg>
    <div
      className="data-title"
    >
      <h3 />
      <p>
        <span />
        <span>
          Countries
        </span>
      </p>
    </div>
  </a>
</div>
`;
